src/Nethermind/Nethermind.Optimism/OPL1CostHelper.cs

```diff
@@ -37,11 +44,32 @@ public UInt256 ComputeL1Cost(Transaction tx, BlockHeader header, IWorldState wor
         if (dataGas == 0)
             return UInt256.Zero;
 
-        UInt256 l1BaseFee = new(worldState.Get(_l1BaseFeeSlot), true);
-        UInt256 overhead = new(worldState.Get(_overheadSlot), true);
-        UInt256 scalar = new(worldState.Get(_scalarSlot), true);
 
-        return ((UInt256)dataGas + overhead) * l1BaseFee * scalar / 1_000_000;
+        if (_opConfigHelper.IsEcotone(header))
+        {
```

Comment:
Should we extract the code inside if and else to separate functions? e.g. `ComputeL1CostEcotone` and `ComputeL1CostPreEcotone`